File: 125-Valid Palindrome/solution.js
//Brute force Approach
function isPalindromeBruteForce(s) {
  s = s.replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
  let reversed = s.split("").reverse().join("");
  return s === reversed;
}

// Two Pointer Approach
function isPalindromeTwoPointer(s) {
  let left = 0,
    right = s.length - 1;

  while (left < right) {
    while (left < right && !isAlphanumeric(s[left])) left++;
    while (left < right && !isAlphanumeric(s[right])) right--;

    if (s[left].toLowerCase() !== s[right].toLowerCase()) return false;
    left++;
    right--;
  }

  return true;
}

// function to check if a character is alphanumeric
function isAlphanumeric(char) {
  return /[a-zA-Z0-9]/.test(char);
}

File: 125-Valid Palindrome/solution.py
import re

# Brute Force Approach
def is_palindrome_brute_force(s):
    s = re.sub(r'[^a-zA-Z0-9]', '', s).lower()
    return s == s[::-1]

# Two Pointer Approach
def is_palindrome_two_pointer(s):
    left, right = 0, len(s) - 1

    while left < right:
        while left < right and not s[left].isalnum():
            left += 1
        while left < right and not s[right].isalnum():
            right -= 1

        if s[left].lower() != s[right].lower():
            return False
        left += 1
        right -= 1

    return True